studio/components/salariesInput/utils/parseSalaries.tsx

```diff
@@ -0,0 +1,83 @@
+import { Result, ResultError, ResultOk } from "../../../utils/result";
+
+export interface Salaries {
+  [year: string]: number;
+}
+
+const NON_EMPTY_DIGITS_ONLY_REGEX = new RegExp(/^\d+$/);
+export const VALID_SALARY_REGEX = NON_EMPTY_DIGITS_ONLY_REGEX;
+
+export enum SalariesParseErrorType {
+  INVALID_SHAPE,
+  INVALID_DATA,
+  NO_DATA,
+  INVALID_FORMAT,
+}
+
+export type SalariesParseError =
+  | {
```

Comment:
this is only a suggestion
```suggestion
  | invalidData | missingData
```